chore: uppercase the liquibase table names

jkwatson-verta opened the pull request:
This is to account for a bug in liquibase when working with an H2 database.

<!-- Title of the PR must comply with [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) guidelines. --> 
<!-- Title should include the JIRA ticket in square brackets after the conventional commit prefix. This will automatically link the PR in JIRA. -->
<!-- Example: "fix: [JIRA-123] Allow creation of groups with no members" -->

## Impact and Context
<!-- Brief description of why these changes are necessary. Don't rewrite the entire Jira ticket. -->

## Risks and Area of Effect
<!--
  Describe both risk (how likely this is to break) and area of effect (how wide potential breakages could reach).
  These should be smaller scale than those documented in a design doc.
-->

## Testing
<!-- Explain how this contribution has been tested, e.g. what tests were added and where they have been run. -->

## How to Revert
<!-- List steps required to revert this change. For example, note if we'd need to revert liquibase changes. -->

jkwatson-verta committed:
chore: uppercase the liquibase table names
This is to account for a bug in liquibase when working with an H2 database.

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/18/

eheinlein-verta reviewed (commented):
There is code in UAC that expects the table name to be lowercased (https://github.com/VertaAI/verta-uacservice/blob/013d8424b06144d7968b766a9957b4b5c847e612/src/main/java/ai/verta/uac/App.java#L644) specifically because we switched from the old name to the new one that was being used in modeldb. Does this take that into account? (By which I mean has it been tested in UAC, and if need be is there a separate PR for UAC?)

jkwatson-verta commented:
> There is code in UAC that expects the table name to be lowercased (https://github.com/VertaAI/verta-uacservice/blob/013d8424b06144d7968b766a9957b4b5c847e612/src/main/java/ai/verta/uac/App.java#L644) specifically because we switched from the old name to the new one that was being used in modeldb. Does this take that into account? (By which I mean has it been tested in UAC, and if need be is there a separate PR for UAC?)

Tables names are not supposed to be case-sensitive (you can always query against either case in every RDBMS I've ever come across), which is why I believe there is a bug in liquibase about this. If we're dependent on table name casing, then we also have a bug.

It looks like UAC already deals with this, though: https://github.com/VertaAI/verta-uacservice/blob/main/src/main/java/ai/verta/uac/App.java#L599-L601

jkwatson-verta commented:
!rebuild

jkwatson-verta commented:
!rebuild

jkwatson-verta committed:
make the upper casing only happen for H2 databases

jkwatson-verta committed:
Merge remote-tracking branch 'origin/jw/uppercase_liquibase' into jw/uppercase_liquibase

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/25/

jkwatson-verta commented:
> > There is code in UAC that expects the table name to be lowercased (https://github.com/VertaAI/verta-uacservice/blob/013d8424b06144d7968b766a9957b4b5c847e612/src/main/java/ai/verta/uac/App.java#L644) specifically because we switched from the old name to the new one that was being used in modeldb. Does this take that into account? (By which I mean has it been tested in UAC, and if need be is there a separate PR for UAC?)
> 
> Tables names are not supposed to be case-sensitive (you can always query against either case in every RDBMS I've ever come across), which is why I believe there is a bug in liquibase about this. If we're dependent on table name casing, then we also have a bug.
> 
> It looks like UAC already deals with this, though: https://github.com/VertaAI/verta-uacservice/blob/main/src/main/java/ai/verta/uac/App.java#L599-L601

I decided to verify this worked before I merged, and good thing I did. It looks like Amazon Aurora actually has case-sensitive table names, so this totally broke things. I updated it to only uppercase when H2 is the database. Thanks for spotting that, @eheinlein-verta , as that would have been a really bad thing to merge!

jkwatson-verta committed:
streamline the code a bit

jkwatson-verta committed:
Merge remote-tracking branch 'origin/jw/uppercase_liquibase' into jw/uppercase_liquibase

# Conflicts:
#	backend/common/src/main/java/ai/verta/modeldb/common/CommonDBUtil.java

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/27/

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/CommonDBUtil.java:
```diff
@@ -288,9 +287,16 @@ protected void runLiquibaseMigration(
+      // H2 upper cases all table names, and liquibase has issues if you don't make this also upper
+      // case.
+      changeLogLockTableName = changeLogLockTableName.toUpperCase();
+      changeLogTableName = changeLogTableName.toUpperCase();
```
```suggestion
      changeLogTableName = changeLogTableName.toUpperCase();
      changeLogLockTableName = changeLogLockTableName.toUpperCase();
```

jkwatson-verta committed:
Update backend/common/src/main/java/ai/verta/modeldb/common/CommonDBUtil.java
